generic_server.cc

```diff
@@ -115,15 +113,15 @@ server::~server()
 }
 
 future<> server::stop() {
-    if (!_stopping) {
+    if (!_gate.is_closed()) {
```

Comment:
I'd move it to inside `shutdown()` in case it is called from other paths,
but can be done as a followup as this hunk follow the existing logic with that respect.

Reply:
> I'd move it to inside `shutdown()` in case it is called from other paths, but can be done as a followup as this hunk follows the existing logic with that respect.

Apologies, I don't understand -- what exactly should I move into `shutdown()`? Thanks.

Reply:
The condition.
I think it'd be better to check `if (!_gate.is_closed())` in `shutdown()` rather than here to allow calling it from other places as well.

Reply:
done in v4